server/server.go

```diff
@@ -875,12 +838,12 @@ func buildServerTimeouts(globalConfig configuration.GlobalConfiguration) (readTi
 	return readTimeout, writeTimeout, idleTimeout
 }
 
-func (s *Server) buildEntryPoints(globalConfiguration configuration.GlobalConfiguration) map[string]*serverEntryPoint {
+func (s *Server) buildEntryPoints() map[string]*serverEntryPoint {
 	serverEntryPoints := make(map[string]*serverEntryPoint)
-	for entryPointName := range globalConfiguration.EntryPoints {
-		router := s.buildDefaultHTTPRouter()
+	for entryPointName := range s.entryPoints {
+		rt := s.buildDefaultHTTPRouter()
```

Comment:
I think that you could directly use `s.buildDefaultHTTPRouter()` in the `middlewares.NewHandlerSwitcher` function instead of creating a variable.